provider/lxd/environ.go

```diff
@@ -104,11 +105,36 @@ func (env *environ) profileName() string {
 	return "juju-" + env.Name()
 }
 
+type LXDProfiler interface {
+	MaybeWriteLXDProfile(pName string, put *charm.LXDProfile) error
+	Name() string
```

Comment:
So I think we can drop the name if the charm name and revision are unique enough.

Reply:
The challenge is the case were the same charm/revision is deployed in to different models.  Howe do we differentiate?

Reply:
dropping Name(), found a different way to get the info in provisioninginfo.go